### PR TITLE
security considerations drafted by Harald

### PR DESCRIPTION
**Pull request by dontcallmedom**
see also https://www.w3.org/Bugs/Public/show_bug.cgi?id=20806

### CODE REVIEW

**Comment by alvestrand** [comment]
@ekr tagged. Will look at better language for "the fact of communication".

File: webrtc.html [hunk 1]
```diff
@@ -5024,10 +5024,71 @@ <h3 id="sec.identity-verify-assertion">Verifying Identity Assertions</h3>
...
+    the Web model against sending data between entities with different
+    origins.</p>
+
+    <p>The WebRTC specification provides no user prompts or chrome
+    indicators for communication; it assumes that once the Web page
+    has been allowed to access media, it is free to share that media
+    with other entities as it chooses.</p>
+
+    <p>A mechanism, <code>peerIdentity</code>, is provided that gives
+    Javascript the option of requesting media that the Javascript
```
Comment (ekr) [inline webrtc.html]: s/the Javascript/that same javascript/

File: webrtc.html [hunk 2]
```diff
@@ -5024,10 +5024,71 @@ <h3 id="sec.identity-verify-assertion">Verifying Identity Assertions</h3>
...
+    cannot access, but can only be sent to certain other entities.</p>
+
+    <p>While a web application already before WebRTC will know the IP
+    address to which the application is delivered, setting up
```
Comment (ekr) [inline webrtc.html]: I think you want to say something like "Even without WebRTC, a Web application will know..."

File: webrtc.html [hunk 3]
```diff
@@ -5024,10 +5024,71 @@ <h3 id="sec.identity-verify-assertion">Verifying Identity Assertions</h3>
...
+    communications exposes additional information about the browser’s
+    network context to the web application, including the set of IP
+    addresses available to the browser for WebRTC use. Some of this
+    information has to be passed to the corresponding party to enable
+    the establishment of a communication session.</p>
+
+    <p>Revealing IP addresses can leak location and means of
```
Comment (ekr) [inline webrtc.html]: I think this should be clearer about the distinction between the counterparty and the server.

As you say, the server knows the public IP but not the private IP (including IPs between VPNs).

The counterparty knows none of these...

File: webrtc.html [hunk 4]
```diff
@@ -5024,10 +5024,71 @@ <h3 id="sec.identity-verify-assertion">Verifying Identity Assertions</h3>
...
+    connection; this can be sensitive. Some mitigation is possible by
+    using relays (for instance TURN servers) rather than direct
+    connections between participants.</p>
+
+    <p>Since the browser is an active platform executing in a trusted
+    network environment (inside the firewall), it is important to
+    limit the damage that the browser can do to other units on the
```
Comment (ekr) [inline webrtc.html]: s/units/elements/

**Comment by ekr** [comment]
This probably could use another editorial pass but it's good enough to merge, IMO.

File: webrtc.html (hunk 3 above)
Comment (dontcallmedom) [inline webrtc.html]: I didn't quite know how to phrase this; suggestions are welcomed.

**Comment by alvestrand** [comment]
Replaced by PR #194.

**Pull request closed without merging**